linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -777,18 +649,62 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
     }
 
     tree match {
+      // Definitions
+
+      case VarDef(ident, _, vtpe, _, rhs) =>
+        checkDeclareLocalVar(ident)
+        typecheckExpect(rhs, env, vtpe)
+
       // Control flow constructs
 
-      case Block(statsAndExpr) =>
-        val stats :+ expr = statsAndExpr
-        val envAfterStats = stats.foldLeft(env) { (prevEnv, stat) =>
-          typecheckStat(stat, prevEnv)
-        }
-        typecheckExpr(expr, envAfterStats)
+      case Skip() =>
+
+      case Block(trees) =>
+        if (trees.isEmpty)
+          reportError("illegal empty block")
```

Comment:
This check is unnecessary. A `Block` will fail to be created with no trees as it cannot determine it's type.